static/compiler-service.ts

```diff
@@ -113,9 +105,11 @@ export class CompilerService {
                     }
                     return null;
                 });
+
                 // Ensure that if no compiler is present, we return null instead of undefined
                 return matchingCompilers.find(compiler => compiler !== null) ?? null;
             }
+
```

Comment:
might still need to return
```
                     return {
                        // There were no compilers, so return null, the selection will show up empty
                        compiler: null,
                        langId: langId,
                    };
```
in case langId is valid...